Kernel/Devices/Audio/AC97.cpp

```diff
@@ -78,13 +77,16 @@ bool AC97::handle_irq(RegisterState const&)
         | AudioStatusRegisterFlag::FIFOError;
     pcm_out_status_register.out(pcm_out_status);
 
+    // If there is no buffer completion, we're not going to do anything
+    if (!is_completion_interrupt)
+        return true;
```

Comment:
You should actually return `false` here if the AC97 device was not responsible for the IRQ.

Reply:
And I should also not reset any interrupt flags in that case as well. Updated, thanks!